autoPyTorch/data/utils.py

```diff
@@ -194,21 +194,21 @@ def reduce_precision(
         For dataframe, the column's precision is reduced using pd.to_numeric.
 
     Args:
-        X:  DatasetCompressionInputType
+        X (DatasetCompressionInputType):
             The data to reduce precision of.
 
     Returns:
         Tuple[DatasetCompressionInputType, DatasetDTypeContainerType, DatasetDTypeContainerType]
             Returns the reduced data X along with the dtypes it and the dtypes it was reduced to.
     """
-    precision: Optional[DatasetDTypeContainerType] = None
+    reduced_dtypes: Optional[DatasetDTypeContainerType] = None
     if isinstance(X, np.ndarray) or issparse(X):
         dtypes = X.dtype
         if X.dtype not in supported_precision_reductions:
             raise ValueError(f"X.dtype = {X.dtype} not equal to any supported"
```

Comment:
We would like to test here using a data frame including categorical features

Reply:
this line is running for numpy arrays, but I can add a test that checks this error.